### PR TITLE
Feature/ch 2988 terraform prod

### PR DESCRIPTION
**Pull request by reshmaattar**

### CODE REVIEW

File: prod/mochi/network/main.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+terraform {
+  backend "s3" {}
+}
+
+module "vpc" {
+  source = "../../../modules/mochi/network"
+
+  infra_env      = var.infra_env
+  vpc_cidr       = "10.0.0.0/17"
```
Comment (sudhir-loveshark) [inline prod/mochi/network/main.tf:14]: I think I missed this originally - why is the netmask here 17 specifically?

File: prod/mochi/network/main.tf [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  azs            = ["eu-west-1a", "eu-west-1b", "eu-west-1c"]
+  public_subnets = slice(cidrsubnets("10.0.0.0/17", 4, 4, 4, 4, 4, 4), 0, 3)
```
Comment (sudhir-loveshark) [inline prod/mochi/network/main.tf:16]: what does the result end up looking like?

File: prod/mochi/redis/variables.tf [hunk 3]
```diff
@@ -0,0 +1,19 @@
+variable "infra_env" {
+  type        = string
+  description = "infrastructure environment"
+
+}
+
+variable "redis_node_type" {
+  type        = string
+  description = "rredis_node_type for redis instance"
```
Comment (sudhir-loveshark) [inline prod/mochi/redis/variables.tf:9]:
```suggestion
  description = "node type for redis instance"
```

File: prod/mochi/sns/variable.tf [hunk 4]
```diff
@@ -0,0 +1,5 @@
+variable "sns_platform_application" {
+  type        = string
+  description = "sns_platform_application is an platform application name"
```
Comment (sudhir-loveshark) [inline prod/mochi/sns/variable.tf:3]:
```suggestion
  description = "sns platform application name"
```

File: prod/mochi/network/main.tf (hunk 2 above)
Comment (reshmaattar) [reply prod/mochi/network/main.tf:16]: its automatically slice up cidr block list of ips for each subnet

File: staging/mochi/variables.tfvars [hunk 5]
```diff
@@ -1,8 +1,9 @@
...
-sms_monthly_dollar_limit = "200"
\ No newline at end of file
+infra_env                  = "staging"
+mysql_instance_class       = "db.t2.medium"
+rds_cluster_instance_count = "1"
+stream_name                = "loveshark-staging"
+kinesis_shard_count        = "1"
+redis_node_type            = "cache.t2.medium"
+redis_nodes                = "2"
+sns_platform_application   = "mochi-integration-testing-application"
```
Comment (sudhir-loveshark) [inline staging/mochi/variables.tfvars:8]:
```suggestion
sns_platform_application   = "mochi-staging-application"
```

File: prod/mochi/network/main.tf (hunk 1 above)
Comment (reshmaattar) [reply prod/mochi/network/main.tf:14]: you can choose anything 17 or 16 .it gives you list of ip range

File: staging/mochi/variables.tfvars (hunk 5 above)
Comment (reshmaattar) [reply staging/mochi/variables.tfvars:8]: lets keep this as its already deployed

File: prod/mochi/network/main.tf (hunk 2 above)
Comment (sudhir-loveshark) [reply prod/mochi/network/main.tf:16]: what's the output? what IP range do they cover?

Comment (reshmaattar) [reply prod/mochi/network/main.tf:16]: https://stackoverflow.com/questions/67360045/how-to-pass-cidr-range-for-subnets-using-cidrsubnets-to-vpc-module-in-terraform

**Pull request merged**